### PR TITLE
Fix workflow deployment verification and add logs

### PR DESCRIPTION
**Pull request by vinumaddumage**
#### Purpose
Fix intermittent Integration test failure on UserMeApproval test.

- Verification of workflow deployment
- Task creation on user addition.

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java [hunk 1]
```diff
@@ -200,50 +202,66 @@ protected void addAssociationForMatch() throws Exception {
...
+                    log.info("BPEL Workflow deployed successfully in " + count + " attempt(s).");
                 }
             }
 
-            // Wait 20 times for 5 seconds intervals until the workflow is properly deployed. This is for server based
-            // test runners like Jenkins.
-            if (count < 20) {
-                log.info("Still no luck :(. So going to wait 5 seconds and try " + (20 - count) + " more time(s).");
-                Thread.sleep(5000);
+            if(isBpelDeployed) {
```
Comment (pulasthi7) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java:220]: fix formatting

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java [hunk 2]
```diff
@@ -200,50 +202,66 @@ protected void addAssociationForMatch() throws Exception {
...
+
+        // Give up after 100 seconds.
+        if (!isBpelDeployed || !isHumanTaskDeployed) {
+            log.info("No luck. Going to give up. Test will most probably fail.");
+        }
+        log.info("Waiting for 5 seconds before proceeding.");
+        Thread.sleep(5000);
+    }
+
+    private boolean isWorkflowDeployed(String url) throws IOException {
```
Comment (pulasthi7) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java:242]:
```suggestion
    private boolean isServiceDeployed(String url) throws IOException {
```

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java [hunk 3]
```diff
@@ -200,50 +202,66 @@ protected void addAssociationForMatch() throws Exception {
...
+        String url;
+        for (int count = 1; count <= MAX_WAIT_ITERATIONS_TILL_WORKFLOW_DEPLOYMENT; count++) {
+            log.info("Verifying workflow deployment on " + count + " of " +
+                    MAX_WAIT_ITERATIONS_TILL_WORKFLOW_DEPLOYMENT + " attempts.");
+            if (!isBpelDeployed) {
+                log.info("Verifying BPEL deployment.");
+                url = identityContextUrls.getSecureServiceUrl() + "/" + addUserWorkflowName + "Service?wsdl";
+                if (isWorkflowDeployed(url)) {
+                    isBpelDeployed = true;
+                    log.info("BPEL Workflow deployed successfully in " + count + " attempt(s).");
```
Comment (pulasthi7) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java:216]: better reword the log. we are not deploying it here. we are just checking for the deployment completion.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java [hunk 4]
```diff
@@ -200,50 +202,66 @@ protected void addAssociationForMatch() throws Exception {
...
+                    break;
+                }
             }
+            log.info("Still no luck :(. So going to wait 1 seconds and retry");
+            Thread.sleep(1000);
+        }
+
+        // Give up after 100 seconds.
+        if (!isBpelDeployed || !isHumanTaskDeployed) {
+            log.info("No luck. Going to give up. Test will most probably fail.");
```
Comment (pulasthi7) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java:236]: this should be a warn log (or error)

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java [hunk 5]
```diff
@@ -200,50 +202,66 @@ protected void addAssociationForMatch() throws Exception {
...
+        }
+        log.info("Waiting for 5 seconds before proceeding.");
+        Thread.sleep(5000);
```
Comment (pulasthi7) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java:239]: why we need 5 more sec after the deployment?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java [hunk 6]
```diff
@@ -200,50 +202,66 @@ protected void addAssociationForMatch() throws Exception {
...
+            log.info("Response content: " + line);
+            if (line.contains("The service cannot be found for the endpoint reference") ||
+                    line.contains("The endpoint reference (EPR) for the Operation not found")) {
+                break;
             }
-            count++;
-        } while (runLoop);
+        }
+
+        return (httpResponse.getStatusLine().getStatusCode() == 200);
```
Comment (pulasthi7) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java:264]: shouldn't the status code be checked before and returned? we would not need to read the body if the expected code is not received

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java (hunk 1 above)
Comment (vinumaddumage) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java]: addressed by af0f52a25c490ea7670ae5b3a8b7628cc45b903a

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java (hunk 2 above)
Comment (vinumaddumage) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java]: updated by af0f52a25c490ea7670ae5b3a8b7628cc45b903a

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java (hunk 3 above)
Comment (vinumaddumage) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java]: addressed by af0f52a25c490ea7670ae5b3a8b7628cc45b903a

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java (hunk 6 above)
Comment (vinumaddumage) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java]: addressed by af0f52a25c490ea7670ae5b3a8b7628cc45b903a

**Review by pulasthi7 — APPROVED** [review]

**Pull request merged**